src/transformers/models/instructblip/modeling_instructblip.py

```diff
@@ -1278,7 +1278,7 @@ def __init__(self, config: InstructBlipConfig):
             self._no_split_modules.extend(language_model._no_split_modules)
 
         if language_model._keep_in_fp32_modules is not None:
-            self._keep_in_fp32_modules.extend(language_model._keep_in_fp32_modules)
+            self._keep_in_fp32_modules_in_instance = language_model._keep_in_fp32_modules
```

Comment:
use `_keep_in_fp32_modules_in_instance` to store the information from the component. This instance attribute will be used in `_keep_in_fp32_full_modules` to get the full list of fp32 module list.